cn/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md

```diff
@@ -0,0 +1,210 @@
+---
+title: 通过文件将Pod信息呈现给容器
+---
+
+{% capture overview %}
+
+此页面描述如何使用DownwardAPIVolumeFile把自己的信息呈现给pod中运行的容器。DownwardAPIVolumeFile可以呈现pod的字段和容器字段。
+
+{% endcapture %}
+
+
+{% capture prerequisites %}
+
+{% include task-tutorial-prereqs.md %}
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Downward API
+
+有两种方式可以将Pod和Container字段呈现给运行中的容器：
+
+* [环境变量](/docs/tasks/configure-pod-container/environment-variable-expose-pod-information/)
+* DownwardAPIVolumeFiles
+
+这两种呈现Pod和Container字段的方式都称为*Downward API*。
+
+## 存储Pod字段
+
+在这个练习中，你将创建一个包含一个容器的pod。这是该pod的配置文件：
+
+{% include code.html language="yaml" file="dapi-volume.yaml" ghlink="/docs/tasks/inject-data-application/dapi-volume.yaml" %}
+
+在配置文件中，你可以看到Pod有一个`downwardAPI`类型的Volume，并且挂载到容器中的`/etc`。
+
+查看`downwardAPI`下面的`items`数组。每个数组元素都是一个[DownwardAPIVolumeFile](/docs/resources-reference/{{page.version}}/#downwardapivolumefile-v1-core)。
+第一个元素指定Pod的`metadata.labels`字段的值保存在名为`labels`的文件中。
+第二个元素自定Pod的`annotations`字段的值保存在名为`annotations`的文件中。
+
+**注意:** 本示例中的字段是Pod字段，不是Pod中容器的字段。 
+{: .note}
+
+创建Pod：
+
+```shell
+kubectl create -f https://k8s.io/docs/tasks/inject-data-application/dapi-volume.yaml
+```
+
+验证Pod中的容器运行正常：
+
+```shell
+kubectl get pods
+```
+
+查看容器的日志：
+
+```shell
+kubectl logs kubernetes-downwardapi-volume-example
+```
+
+输出显示`labels`和`annotations`文件的内容：
+
+```shell
+cluster="test-cluster1"
+rack="rack-22"
+zone="us-est-coast"
+
+build="two"
+builder="john-doe"
+```
+
+进入Pod中运行的容器，打开一个shell：
+
+```
+kubectl exec -it kubernetes-downwardapi-volume-example -- sh
+```
+
+在该shell中，查看`labels`文件：
+
+```shell
+/# cat /etc/labels
+```
+
+输出显示Pod的所有labels都已写入`labels`文件。
+
+```shell
+cluster="test-cluster1"
+rack="rack-22"
+zone="us-est-coast"
+```
+
+同样，查看`annotations`文件：
+
+```shell
+/# cat /etc/annotations
+```
+
+查看`/etc`目录下的文件：
+
+```shell
+/# ls -laR /etc
+```
+
+在输出中可以看到，`labels` 和 `annotations`文件都在一个临时子目录中：这个例子，`..2982_06_02_21_47_53.299460680`。在`/etc`目录中，`..data`是一个指向临时子目录
+的符号链接。`/etc`目录中，`labels` 和 `annotations`也是符号链接。
+
+```
+drwxr-xr-x  ... Feb 6 21:47 ..2982_06_02_21_47_53.299460680
+lrwxrwxrwx  ... Feb 6 21:47 ..data -> ..2982_06_02_21_47_53.299460680
+lrwxrwxrwx  ... Feb 6 21:47 annotations -> ..data/annotations
+lrwxrwxrwx  ... Feb 6 21:47 labels -> ..data/labels
+
+/etc/..2982_06_02_21_47_53.299460680:
+total 8
+-rw-r--r--  ... Feb  6 21:47 annotations
+-rw-r--r--  ... Feb  6 21:47 labels
+```
+
+用符号链接可实现元数据的动态原子刷新；更新将写入一个新的临时目录，然后`..data`符号链接完成原子更新，通过使用[rename(2)](http://man7.org/linux/man-pages/man2/rename.2.html)。
+
+退出shell：
+
+```shell
+/# exit
+```
+
+## 存储容器字段
+
+前面的练习中，你将Pod字段保存到DownwardAPIVolumeFile中。接下来这个练习，你将存储容器字段。这里是包含一个容器的pod的配置文件：
+
+{% include code.html language="yaml" file="dapi-volume-resources.yaml" ghlink="/docs/tasks/inject-data-application/dapi-volume-resources.yaml" %}
```

Comment:
copy a dapi-volume-resources.yaml to /cn/docs/tasks/inject-data-application/dapi-volume-resources.yaml